### PR TITLE
feat: Add Multipart HTTP Signer support

### PR DESCRIPTION
**Pull request by abyss-w**

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -530,6 +469,141 @@ func (s *Storage) nextPartPage(ctx context.Context, page *PartPage) error {
...
+	return
+}
+
+func (s *Storage) querySignHTTPDelete(ctx context.Context, path string, expire time.Duration, opt pairStorageQuerySignHTTPDelete) (req *http.Request, err error) {
+	pairs, err := s.parsePairStorageDelete(opt.pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	if pairs.HasMultipartID {
```
Comment (JinnyYi) [inline storage.go:533]: Maybe we should also support `ObjectMode` to delete dir?

Comment (abyss-w) [reply storage.go:533]: We must use `AbortMultipart` to remove the `multipart object` before `completeMultipart`, in this case we must use `multipartID`.

Comment (abyss-w) [reply storage.go:533]: Maybe you're right.

Comment (abyss-w) [reply storage.go:533]: Oh, I added it, in [utils.go](https://github.com/beyondstorage/go-service-s3/pull/237/files?authenticity_token=mdOfA5eA5gsgNOoesXvE8DFlLiMmL9vWIxv9khbU8ap5PeWudO7VPWg%2F4QXljW0kdUDcR6wSfugQW1rqKBuAtg%3D%3D&file-filters%5B%5D=.go&file-filters%5B%5D=.mod#diff-8b7f25bac2ae04c2ce2abb628c6aa28103fd113103a62f3e9d232fa995684915R447).

Comment (abyss-w) [reply storage.go:533]: Maybe I shouldn't put it in `utils.go`.

Comment (JinnyYi) [reply storage.go:533]: Oh, I missed it.

**Review by JinnyYi — APPROVED** [review]

**Pull request merged**